src/network/core/tcp_coordinator.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file tcp_coordinator.h Basic functions to receive and send TCP packets to/from the Game Coordinator server.
+ */
+
+#ifndef NETWORK_CORE_TCP_COORDINATOR_H
+#define NETWORK_CORE_TCP_COORDINATOR_H
+
+#include "os_abstraction.h"
+#include "tcp.h"
+#include "packet.h"
+#include "game_info.h"
+
+/** Enum with all types of TCP Game Coordinator packets. The order MUST not be changed. **/
+enum PacketCoordinatorType {
```

Comment:
I think it would be useful to add some extra documentation as here there are two concepts getting intermingled. You have the game server, game client and game coordinator, and the protocol server and client. The game coordinator being the server in the protocol is right, though you get things like CLIENT_REGISTER which is the server registering to the coordinator. What about:

There are three types of entities in this protocol. The game coordinator to which game servers register and game clients request the list of servers, the game server that host the games that are played and the game clients that are the end users playing the games. In this protocol the game coordinator is seen as the server (everything connects to the game coordinator) and as such packets sent by the game coordinator have SERVER in the type name. The game server and game client connect to the game server and as such packets sent by them have the CLIENT in the type name.

Next to that I would prefix server and client in the comments below with game to make the distinction between the protocol client/server and the game client/server.